stdlib/public/core/String.swift

```diff
@@ -15,22 +15,24 @@ import SwiftShims
 /// A type that can represent a string as a collection of characters.
 public protocol StringProtocol
   : RangeReplaceableCollection, BidirectionalCollection,
-  CustomDebugStringConvertible,
-  CustomReflectable, CustomPlaygroundQuickLookable,
   TextOutputStream, TextOutputStreamable,
   LosslessStringConvertible, ExpressibleByStringLiteral,
   Hashable, Comparable
   where Iterator.Element == Character {
 
-  associatedtype UTF8Index
-  var utf8: String.UTF8View { get }
-  associatedtype UTF16Index
-  var utf16: String.UTF16View { get }
-  associatedtype UnicodeScalarIndex
-  var unicodeScalars: String.UnicodeScalarView { get }
-  /*associatedtype CharacterIndex*/
-  var characters: String.CharacterView { get }
+  associatedtype UTF8View : /*Bidirectional*/Collection
+  where UTF8View.Element == UInt8 // Unicode.UTF8.CodeUnit
```

Comment:
Since these `where` clauses are a new thing. We never really agreed on the indentation rules. I would prefer them to be indented.
Also, why can't you use `Unicode.UTF8.CodeUnit` instead of `UInt8`?